### DISCUSSION TITLE
improve: 在引导文档中讲解文件夹的作用，哪些目录是可以删除的，以及如何配置为自己喜欢的样子

### DISCUSSION THREAD

**Discussion by nekomeowww** [nolebase/nolebase#157] [Q&A]
Inspired from #11 .

**Comment by Lionad-Morotar** [comment]
很喜欢 nolebase！我想使用 nolebase 把自己的 VuePress 0.x 博客升级到博客 plus plus 版本（T。T）但是遇到了一个实际的问题：如何在不丢掉原有博客的 Git 提交记录且保证能跟上 nolebase 主项目更新节奏的情况下进行升级？换句话说，假使 nolebase 是一个脚手架工具，而不仅仅作为一个模板项目，我就能很方便地使用它。关于这点大佬们有何看法（meow

**Comment by nekomeowww** [comment]
> 如何在不丢掉原有博客的 Git 提交记录且保证能跟上 nolebase 主项目更新节奏的情况下进行升级？

这是一个好问题！

我们在 7 月份的时候就有过这样的困境，有伙伴无法很好的更新 Nolebase（更何况 Nolebase 只是一个基于 VitePress 的封装）

我们现在正在把所有的插件和组件拆分到 [Nolebase Integrations](https://github.com/nolebase/integrations) （你也可以在这里看到 [Nolebase Integrations 的文档](https://nolebase-integrations.ayaka.io/pages/en/)）

我想，如果你想要直接使用我们封装过的组件和工具的话，不妨看看 Nolebase Integrations 先把博客直接迁移到 VitePress 然后装一些 Nolebase 也有的插件和 Nolebase 自己的插件实现部分这样的功能？

> 换句话说，假使 nolebase 是一个脚手架工具，而不仅仅作为一个模板项目，我就能很方便地使用它。

这是一个很好的建议和想法！

真的很抱歉我们现在的知识库耦合得非常严重 🥺

原本我是想把 Nolebase 当作是一个 VitePress 的主题去对待的，但是后来发现很多 markdown 的预处理和 HTML 的后处理都会涉及到非主题相关的操作，而主题似乎没办法很好的利用到 Vite 的能力去做很多转换和处理的操作。我们现在初步的想法是想要先把 Nolebase 具备的特殊能力拆分到 Nolebase Integrations 这样的子项目里面，然后把 Nolebase 单独独立出来做一个模板项目，大家点 GitHub 的 use this template 就能直接用的那种（也可以进一步支持 #13 ，一键部署到 Netlify 和 Vercel，甚至实现直接从 Obsidian 部署的能力），然后再考虑应该如何继续前行。

不知道这样的解答是否能给解开你的疑惑呢？

@Lionad-Morotar

**Comment by Lionad-Morotar** [comment]
好耶！Nolebase Integrations 是个折中的选择，我会尝试去使用它的，感谢~

**Comment by LemonNekoGH** [comment]
按照我目前的理解：
- `笔记/` 文件夹是所有笔记文档的根目录，在修改这个文件夹名称时，需要同时修改
https://github.com/nolebase/nolebase/blob/f043eff26b98752c5a61f3ebf0d828de0f48c355/scripts/update.ts#L15
中 `target` 的值，这样在运行 `pnpm docs:dev` 等命令的时候可以正常生成侧边栏目录。

**Comment by zll600** [comment]
我理解修改这里，可以修改首页的顶部的导航栏（不知道描述的对不对） https://github.com/nolebase/nolebase/blob/b5903bffb0253087c3139259a9d48885f9ed08fe/.vitepress/config.ts#L214

**Comment by nekomeowww** [comment]
> 我理解修改这里，可以修改首页的顶部的导航栏（不知道描述的对不对）
> 
> https://github.com/nolebase/nolebase/blob/b5903bffb0253087c3139259a9d48885f9ed08fe/.vitepress/config.ts#L214

可以的，理解的对。与此同时还要改一下 scripts/update.ts 的 `笔记`

**Comment by VacuolePaoo** [comment]
> > 如何在不丢掉原有博客的 Git 提交记录且保证能跟上 nolebase 主项目更新节奏的情况下进行升级？
> 
> 这是一个好问题！
> 
> 我们在 7 月份的时候就有过这样的困境，有伙伴无法很好的更新 Nolebase（更何况 Nolebase 只是一个基于 VitePress 的封装）
> 
> 我们现在正在把所有的插件和组件拆分到 [Nolebase Integrations](https://github.com/nolebase/integrations) （你也可以在这里看到 [Nolebase Integrations 的文档](https://nolebase-integrations.ayaka.io/pages/en/)）
> 
> 我想，如果你想要直接使用我们封装过的组件和工具的话，不妨看看 Nolebase Integrations 先把博客直接迁移到 VitePress 然后装一些 Nolebase 也有的插件和 Nolebase 自己的插件实现部分这样的功能？
> 
> > 换句话说，假使 nolebase 是一个脚手架工具，而不仅仅作为一个模板项目，我就能很方便地使用它。
> 
> 这是一个很好的建议和想法！
> 
> 真的很抱歉我们现在的知识库耦合得非常严重 🥺
> 
> 原本我是想把 Nolebase 当作是一个 VitePress 的主题去对待的，但是后来发现很多 markdown 的预处理和 HTML 的后处理都会涉及到非主题相关的操作，而主题似乎没办法很好的利用到 Vite 的能力去做很多转换和处理的操作。我们现在初步的想法是想要先把 Nolebase 具备的特殊能力拆分到 Nolebase Integrations 这样的子项目里面，然后把 Nolebase 单独独立出来做一个模板项目，大家点 GitHub 的 use this template 就能直接用的那种（也可以进一步支持 #13 ，一键部署到 Netlify 和 Vercel，甚至实现直接从 Obsidian 部署的能力），然后再考虑应该如何继续前行。
> 
> 不知道这样的解答是否能给解开你的疑惑呢？
> 
> @Lionad-Morotar

把Nolebase作为一个子项目，git add submodule，作为子仓库
